app/lib/bk/compat/parsers/circleci/orbs.rb

```diff
@@ -28,6 +19,55 @@ def translator(action, _config)
           ]
         end
       end
+
+      # Docker orb specific implementation
+      class DockerOrb
+        def initialize
+          @config = {}
+        end
+
+        def matcher(orb, _conf)
+          orb.start_with?('docker/')
+        end
+
+        def translator(action, config)
+          @config = config
+          method_map = {
+            'docker/build' => -> { translate_docker_build(config) },
+            'docker/install-docker' => -> { validate_and_translate_install_docker(config) }
+          }
+          method_map.fetch(action, -> { ["# Unsupported docker orb action: #{action}"] }).call
+        end
+
+        private
+
+        def validate_and_translate_install_docker(_config)
+          install_dir = @config['install-dir'] || '/usr/local/bin'
+          version = @config['version'] || 'latest'
+          commands = ['"# No need for install docker, the agent takes care of that"']
+          commands << "echo '~~~ Installing Docker'"
+          commands << "mkdir -p #{install_dir}" if install_dir != '/usr/local/bin'
+          commands += install_commands(version, install_dir)
+          commands
```

Comment:
You are creating a single array to return and operating on it 3 different ways. This same thing can be achieved with structures [similar to others used across this codebase](https://github.com/buildkite/migration/blob/main/app/lib/bk/compat/parsers/circleci/steps.rb#L68-L76):

```suggestion
          [
            '# No need for install docker, the agent takes care of that',
            "echo '~~~ Installing Docker'",
            install_dir != '/usr/local/bin' ? "mkdir -p #{install_dir}" : nil,
            install_commands(version, install_dir)
          ].flatten.compact
```